compiler/rustc_errors/src/json.rs

```diff
@@ -554,6 +555,21 @@ impl DiagnosticSpan {
         suggestion
             .substitutions
             .iter()
+            .filter_map(|substitution| {
+                let mut parts = substitution.parts.clone();
```

Comment:
skip outputting suggestions with overlapping spans to JSON, same check as in lib.rs